zips/zip-0032.rst

```diff
@@ -513,43 +525,119 @@ valid diversifiers.
 The default diversifier for $(\mathsf{sk}_i, \mathsf{c}_i)$ is defined to be $d_{i,0}.$
 
 
-Specification: Arbitrary key derivation
-=======================================
+Specification: Registered key derivation
+========================================
+
+In the context of a particular application protocol defined by a ZIP, there is sometimes
+a need to define an HD subtree that will not collide with keys derived for other protocols,
+as far as that is possible to assure by following the ZIP process [#zip-0000]_.
 
-In some contexts there is a need for deriving arbitrary keys with the same derivation path as
-existing key material (for example, deriving an arbitrary account-level key), without the need for
-ecosystem-wide coordination. The following instantiation of the hardened key generation process may
-be used for this purpose.
+Within this subtree, the application protocol may use derivation paths related to those
+used for existing key material — for example, to derive an account-level key. The following
+instantiation of the hardened key generation process may be used for this purpose.
+
+It is strongly RECOMMENDED that implementors ensure that documentation of the usage and
+derivation paths of the application protocol's key tree in the corresponding ZIP is
+substantially complete, before public deployment of software or hardware using this
+mechanism. The ZIP process allows for subsequent updates and corrections.
 
 Let $\mathsf{ContextString}$ be a globally-unique non-empty sequence of at most 252 bytes
 that identifies the desired context.
```

Comment:
Blocking: we need to at a minimum say this SHOULD be specified in the corresponding ZIP. The paragraph above only currently RECOMMENDs documentation of the usage and derivation paths, not the context string(s) used as roots.